tests/Application.IntegrationTests/TestHelpers/IntegrationTestWebApplicationFactory.cs

```diff
@@ -0,0 +1,33 @@
+using Infrastructure.Persistence;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Application.IntegrationTests.TestHelpers;
+
+internal class IntegrationTestWebApplicationFactory : WebApplicationFactory<Program>
+{
+    public DatabaseContainer Database { get; }
+    
+    public IntegrationTestWebApplicationFactory()
+    {
+        Database = new DatabaseContainer();
+    }
+
+    protected override void ConfigureWebHost(IWebHostBuilder webHostBuilder)
+    {
```

Comment:
You could set the connection string to your db container through a command line arg to avoid having it set as an environment variable 

builder.ConfigureAppConfiguration(config =>
        {
            config.AddCommandLine(new[] { $"ConnectionStrings:DefaultConnection={Database.ConnectionString}" });
        });

Reply:
I'll close this issue for now (see my above reply for context) 😊